apps/passport-client/src/password.ts

```diff
@@ -9,3 +16,28 @@ export const checkPasswordStrength = (password: string): boolean => {
 };
 
 export const PASSWORD_MINIMUM_LENGTH = 8;
+
+// For when the user has not set a password yet, and wants to add one
+export const setPassword = async (password: string, dispatch: Dispatcher) => {
+  const crypto = await PCDCrypto.newInstance();
+  const currentEncryptionKey = loadEncryptionKey();
+  const { salt: newSalt, key: newEncryptionKey } =
+    await crypto.generateSaltAndEncryptionKey(password);
+  const res = await updateBlobKeyForEncryptedStorage(
+    currentEncryptionKey,
+    newEncryptionKey,
+    newSalt
+  );
+
+  if (!res.success) {
+    throw new Error(`Request failed with message ${res.error}`);
+  }
+
+  dispatch({
+    type: "change-password",
+    newEncryptionKey,
+    newSalt
+  });
+  // to make sure the salt is uploaded properly
```

Comment:
@artwyman I wonder how this sort of operation interacts with the sync state machine. eg if a sync occurs between lines 26 and 36, it may try to do something that it can't actually do due to the password having changed

Reply:
Definitly a risk.  Changing password is one of the class of blob-sync operations which should ideally happen only one-at-a-time.  I'll have to try to include that when I refactor sync.  For now, I think the only way to avoid that would be to add some sort of a `changingPassword` flag which block sync operations, but it would take more thought to make that fully safe.

In the mean time, I think there are two ways things could go wrong.  A download could try to happen, and would fail because the blob key is missing.  Or an upload could happen and re-create the old blob key, leaving behind 2 of them.  The latter would be the bigger problem.  You could try to avoid that by blocking until `uploadingUploadId` to becomes undefined before allowing change password.  Then optionally doing the reverse and blocking uploads while `changingPassword` is true.  Then you'd need to request a sync after change-password to make sure things start again.  Of course all this blocking has a risk of getting stuck if one of the requests fail.

I expect the whole thing is too complicated to address in this PR and needs the more holistic look I intend to make soon, probably next week.  I don't think this PR is inherently making the situation worse, just drawing attention to it. 
 @rrrliu if you can see a quick version of this that you want to try here, feel free. :)